src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1108,6 +1124,18 @@ private static ObjectType GetParseCidrReturnType()
             }, null);
         }
 
+        private static ObjectType GetParseUriReturnType()
+        {
+            return new ObjectType("parseUri", TypeSymbolValidationFlags.Default, new[]
+            {
+                new TypeProperty("scheme", LanguageConstants.String),
+                new TypeProperty("host", LanguageConstants.String),
+                new TypeProperty("port", LanguageConstants.Int),
+                new TypeProperty("path", LanguageConstants.String),
```

Comment:
I think `path` and `query` should be of type `string | null` since they are optional in the return value. @jeskew mentioned that we can create types like this using something like this:
```csharp
TypeHelper.CreateTypeUnion(<type>, LanguageConstants.Null)
```